setup.py

```diff
@@ -40,7 +40,7 @@
 install_requires += [
     "scipy",
     "jaxtyping==0.2.19",
-    "typeguard~=2.13.3",
+    "typeguard~=4.3.0",
```

Comment:
I'm all for raising the upper bound, but I'm not sure of the benefit of raising the lower bound so high, especially if no other changes were needed - that typeguard version only arrived in May 2024.

I wonder if the right balance of permissibility may look more like `>=2.13.3,<5` or even `>=3,<5`?

(I'm not a maintainer, sorry...)